### DISCUSSION TITLE
Funding and bounty status

### DISCUSSION THREAD

**Discussion by WithoutPants** [stashapp/stash#1699] [Announcements]
Hi folks. This post covers where financial contributions are used, the status of all bounty payments, and ideas for the future of the bounty system.

Let me open by saying thank you to everyone who contributes financially to the stashapp Open Collective (https://opencollective.com/stashapp#category-ABOUT). As noted below, your contributions have enabled focused development of stash and resulted in a number of new features and improvements.

### About me

In lieu of seeking gainful employment, I have been working part-time on stash (and to a lesser extent stash-box) since early last year. Based on the data I've collected, I'm averaging between 12-18 hours per week on development of stash/stash-box and general administration. This is actually less than I'd like due to ongoing care-giving commitments caused by the pandemic.

My intention is to draw an occasional stipend from financial contributions to allow me to continue to dedicate time to stash (and to justify continuing to focus on it in favour of seeking employment). After discussion with the community on Discord in June this year, I claimed $675 from accumulated contributions since the inception of the OpenCollective, on top of bounties that were completed. I am however mindful that others also commit time and effort to the project (@gitgiggety has been contributing a heap of features and improvements lately, for example), so I'd also be seeking feedback from financial and developer contributors as to what is appropriate for me to claim.

### Other costs

We have a ~~$400~~ $600/year expense for maintaining a stash-box instance, and a $10/year expense for the website DNS registration.

### Bounty status (updated 22-Jun-2023)

Transactions from https://opencollective.com/stashapp/transactions

| Transaction no. / date | Amount | Issue | Status |
|--|--|--|--|
| [#663172](https://opencollective.com/stashapp/orders/663172) (split) | $200 | #3829 | Resolved |
| [#663172](https://opencollective.com/stashapp/orders/663172) (split) | $50 | #3815 | Open |
| [#660850](https://opencollective.com/stashapp/orders/660850) | $50 | #3530 | Open |
| [#659259](https://opencollective.com/stashapp/orders/659259) | $10 | #3760 | Pending |
| [#658250](https://opencollective.com/stashapp/orders/658250) | $10 | #3759 | Resolved |
| [#636980](https://opencollective.com/stashapp/orders/636980) | $100 | #3553  | Open |
| 2023-03-14 10:00:00 | $50 | #1259 | Open |
| 2023-03-02 02:27:45 | $10 | **Unknown** | |
| 2023-02-20T09:00:40 | $50 | #3277 | Paid |
| 2023-02-14T03:36:23 | $150 | #3113 | Paid |
| 2023-02-07T05:29:25 | $25 | #1455 | Resolved - transferred to project |
| 2022-12-17T10:50:27 | $100 | #2676 | Paid |
| 2022-12-17T10:21:34 | $100 | #623 | Open |
| 2022-11-27 07:24:30 | $150 | #2844 | Paid |
| 2022-10-21 21:56:07 | $40 | #2497 | Open |
| 2022-10-14 18:42:52 | $100 | #3003 | Open (transferred from #2575/#1444) |
| 2022-09-21 02:06:19 | $50 | #2937 | Paid |
| 2022-08-05 03:40:16 | $50 | #2782 | Transferred to project |
| 2022-06-24 01:32:55 | $35 | **Unknown** |  |
| 2022-06-14 00:50:54 | $10 | **Unknown** |  |
| 2022-06-08 18:42:52 | $100 | #2575/#1444 | Resolved - transferred to #3003 |
| 2022-06-03 09:31:41 | $52.34 | #2643 | Open |
| 2022-05-06 01:30:39 | $30 | #2113 | Paid |
| 2022-04-29 11:21:06 | $50 | #2505 | Paid |
| 2022-04-29 11:18:32 | $50 | #2535 | Paid |
| 2022-03-28 08:11:19 | $50 | #2433 | Paid |
| 2022-03-19 12:42:35 | $20 | #2389 | Paid |
| 2022-03-18 02:29:50 | $10 | [stash-box#282](https://github.com/stashapp/stash-box/issues/282) | Open |
| 2022-03-07 17:31:48 | $251 | #2366 | Paid |
| 2022-03-02 05:22:23 | $50 | Unknown |  |
| 2022-02-23 08:33:32 | $50 | #2311 | Paid |
| 2022-01-16 10:32:37 | $10 | Unknown |  |
| 2021-12-08 12:36:16 | $50 | #2086 | Paid |
| 2021-12-05 07:40:29 | $51.8 | #571 | Open |
| 2021-10-29 18:21:06 | $30 | #1210 | Paid |
| 2021-09-27 04:59:18 | $35 | #1775 | Paid |
| 2021-09-27 00:37:34 | $35 | #1265 | Paid |
| 2021-09-23 02:54:28 | $15 | #1492 | Paid |
| 2021-09-19 15:50:45 | $30 | #1210 | Paid |
| 2021-09-18 09:05:17 | $40 | #1210 | Paid |
| 2021-09-08 03:14:25 | $20 | #571 | Open |
| 2021-09-07 15:33:13 | $65 | #1492 | Paid |
| 2021-08-22 14:02:00 | $20 | #1659 | Open |
| 2021-08-12 09:22:28 | $40 | #1408 | Open |
| 2021-08-12 06:52:04 | $20 | **Unknown** | |
| 2021-08-06 02:48:53 | $20 | **Unknown** | |
| 2021-07-11 03:23:45 | $40 | **Unknown** | |
| 2021-07-05 13:37:57 | $10 | #1556 | Paid |
| 2021-06-11 20:05:19 | $20 | #1492 | Paid |
| 2021-06-11 20:01:38 | $20 | #1493 | Paid |
| 2021-05-25 01:39:29 | $50 | #13 | Open |
| 2021-05-24 11:26:58 | $20 | #1423 | Paid |
| 2021-05-23 00:15:14 | $33 | **Unknown** | |
| 2021-04-11 14:24:44 | $20 | #1282 | Paid | 
| 2021-04-11 14:24:44 | $20 | #1283 | Paid |
| 2021-04-02 15:47:51 | $40 | #1259 | Open |
| 2021-04-02 15:44:12 | $40 | #1258 | Open |
| 2021-04-02 12:19:15 | $40 | #1011 | Paid |
| 2021-03-26 03:47:55 | $25 | #121 | Paid |
| 2021-03-26 03:37:15 | $10 | **Unknown** |
| 2021-03-08 21:21:47 | $10 | #479 | Paid |
| 2021-03-03 12:20:06 | $25 | #557 | Paid |
| 2021-02-24 00:23:04 | $10 | **Unknown** |

#### Summary

| Category | Issues/Transactions | Amount |
|--|--|--|
| Paid | 32 | $1686 |
| Resolved pending payment | 2 | $210 |
| Open | 15 | $714.14 |
| Pending | 1 | $10 |
| **Unassigned** | 11 | $248 |
| Total | 61 | $2868.14 |

If you recognise one of the **Unknown** transactions as your contribution, or if you have donated under the `bounty` tier and your chosen issue has not been assigned the bounty, please contact me on the Discord server, or via OpenCollective so that I can assign it.

### Future of the bounty system

Administering the bounty system is **hard**. OpenCollective doesn't have any built in system for development bounties, and the current system of having the user set their contribution note is problematic for the following reasons:
* I cannot determine the value of a bounty based on the contribution note if the user has contributed before, since the note is shown under the contributors list, and only the total contribution is shown
* it puts the onus on contributors to set the note or otherwise contact the developers to assign the bounty. This results in many bounty tier contributions being made that do not have a bounty assigned since these steps were not followed.

As noted above, we have $248 in unassigned bounties, the oldest being over twelve months old.

I propose the following changes:
* unassigned bounty contributions should be released to the general balance if they are not assigned after a month. The tier description should be amended as such. Existing unassigned bounties would have a month grace period after the change and announcement.
* change the `bounty` description to remove the mention of the issue in the contribution note, since this is not an effective method. Assigning bounties should be instead communicated using OpenCollective or Discord.

Another alternative may be to replace the `bounty` tier with another one-off tier name, and indicate that if the user wishes to assign the contribution to a bounty, then they should contact the devs via OpenCollective or Discord.

Thanks again for your continued support.

**Comment by gitgiggety** [comment]
Only just found this "discussion".

Thanks for the recognition of my work :)

Personally I just started doing some contributions after discovering this project out of my own interests. Both as hobby in working with some techniques which I'm unfamiliar with (Go, React, Typescript) and to either implement some featues I want myself or doing some small fixes here and there to solve some user annoyances. So I'm in no way either looking for funding or prioritizing bounty tickets. So how things are run financially is up to you, and you (or anyone else) shouldn't feel the need to compensate me for my work. For me personally it's more important to have fun in the work I'm doing, and any form of recognition for that is good enough (for which PRs being merged already is sufficient).

**Comment by Privitude** [comment]
Is there a way to contribute to/co-fund a bounty? I'd be willing to double the bounty on #1258 & 1259.

**Reply by WithoutPants** [reply]
Yes. Make the contribution to the OpenCollective under Bounty. If possible add the issue number(s) to the contribution comment, or notify me here or on Discord of the contribution number and I will assign it.

**Reply by Privitude** [reply]
> Yes. Make the contribution to the OpenCollective under Bounty. If possible add the issue number(s) to the contribution comment, or notify me here or on Discord of the contribution number and I will assign it.

Contribution #CONTRIBUTION #635947. Please assign that $50 to #1259. I'll likely contribute more for #1258 at a later time.

**Comment by Privitude** [comment]
You've got #1259 listed twice, once with the original $40 and once with the $50 I added. Could you combine the listings so it shows as $90, please?

**Reply by DogmaDragon** [reply]
It's mentioned on the issue itself <https://github.com/stashapp/stash/issues/1259#issuecomment-1467205648>

**Reply by Privitude** [reply]
> It's mentioned on the issue itself [#1259 (comment)](https://github.com/stashapp/stash/issues/1259#issuecomment-1467205648)

It would be nice if that was reflected here. People are far less likely to read the comments and see that money as added to the bonus than they are to browse a page like this and look for the highest dollar amounts. Right now, #1259 is listed twice, once as $50 and once as $40. The most likely scenario is that people see the first entry, see $50 and move on, then see the second entry, not even realize it's the same issue, and move on. 

Conversely, if they see the entry and then see $90 that might make them pause and investigate. And if no one tries to make it happen at $90, I'll likely increase it further down the line. The more money that is showing on one line, the more people are going to look at it and think "huh, $90/$150/$200 for knocking this out? I could give that a shot!"

**Comment by Privitude** [comment]
So is it not possible to correct the issue of duplicate entries? I'm willing to continue donating more money and upping the bounties on more things, but it's not really worth it if it's going to show up here as two multiple conflicting entries instead of one large bounty. It's the large bounty that's likely to catch someone's eye and motivate them.

**Reply by WithoutPants** [reply]
Advertising bounty amounts is not and has not been the primary purpose of this post. The table posted above maps transactions to their assigned issues. From next month, I will change up the way I report on the funding and bounties, and that should include something about open bounties and their current assigned amounts that should hopefully fulfil the criteria that you're seeking.
